packages/core/src/routes/sign-in-experience/index.ts

```diff
@@ -125,5 +130,51 @@ export default function signInExperiencesRoutes<T extends ManagementApiRouter>(
     }
   );
 
+  router.post(
+    '/sign-in-exp/default/check-password',
+    koaGuard({
+      body: z.object({ password: z.string(), userId: z.string().optional() }),
+      response: z.object({ result: z.literal(true) }).or(
+        z.object({
+          result: z.literal(false),
+          issues: z.array(
+            z.object({ code: z.string(), interpolation: z.record(z.unknown()).optional() })
+          ),
+        })
+      ),
+      status: [200, 400],
+    }),
+    async (ctx, next) => {
+      const { password, userId } = ctx.guard.body;
+      const [signInExperience, user] = await Promise.all([
+        findDefaultSignInExperience(),
+        userId && findUserById(userId),
+      ]);
+      const passwordPolicyChecker = new PasswordPolicyChecker(signInExperience.passwordPolicy);
+
+      const issues = await tryThat(
+        async () =>
+          user
+            ? checkPasswordPolicyForUser(passwordPolicyChecker, password, user)
+            : passwordPolicyChecker.check(password),
+        (error) => {
+          if (error instanceof TypeError) {
+            throw new RequestError('request.invalid_input', { message: error.message });
+          }
+          throw error;
```

Comment:
I think the behavior is a bit different from https://github.com/logto-io/logto/blob/master/packages/core/src/routes/interaction/utils/validate-password.ts#L68-L75
Should we keep them consistent?

Reply:
no it's not the same scenario. can you check changeset?

Reply:
Ah, I see. It makes no sense that the management API should throw in this scenario. OK